### PR TITLE
FISH-99 FISH-298 Schemas are always picked up and used in OpenAPI

### LINKED ISSUE #4802: OpenAPI @Schema implementation is ignored /FISH-298
# Description #
----------

The OpenAPI implementation ignores the implementation value provided in the @Schema annotation.

## Expected Outcome

Resulting http://localhost/openapi document contains a description of the implementation class instead of simply "object".

## Current Outcome

    paths:
      /WS/cdasr/1/dataviews:
        get:
          summary: Gets descriptions of the dataviews that are available from CDAS.
          operationId: getDataviews
          responses:
            "304":
              description: Not Modified
            "406":
              description: Not Acceptable
            default:
              description: Descriptions of all dataviews.
              content:     
                application/json:
                  schema:
                    type: object
                application/xml:
                  schema:
                    type: object

## Steps to reproduce (Only for bug reports) 

Deploy a war containing the following abbreviated JAX-RS application

    @Path("/dataviews")
    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
    @Operation(summary = "Gets descriptions of the dataviews that are " +
                         "available from CDAS.")
    @APIResponse(description = "Descriptions of all dataviews.",
                 content = @Content(mediaType = MediaType.APPLICATION_JSON,
                     schema = @Schema(implementation =  gov.nasa.gsfc.cdaweb.schema.Dataviews.class)))
    @APIResponse(responseCode = "304", description = "Not Modified")
    @APIResponse(responseCode = "406", description = "Not Acceptable")
    public Response getDataviews(

Where gov.nasa.gsfc.cdaweb.schema.Dataviews was generated by xjc compiling https://cdaweb.gsfc.nasa.gov/WebServices/REST/CDAS.xsd.  Here's the resulting code

    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "dataviewDescription"
    })
    @XmlRootElement(name = "Dataviews")
    public class Dataviews {

        @XmlElement(name = "DataviewDescription", required = true)
        protected List<DataviewDescription> dataviewDescription;

       public List<DataviewDescription> getDataviewDescription() {
            if (dataviewDescription == null) {
                dataviewDescription = new ArrayList<DataviewDescription>();
            }
            return this.dataviewDescription;
        }
    }

and the class is in a jar that is inside the application's war (in WEB-INF/lib).

## Context (Optional)

I was hoping that by providing the class in the @APIResponse/@Content/@Schema annotations, the resulting openapi document would contain a more details/useful schema/type value than simply "object".

## Environment ##

- **Payara Version**: 5.2020.2
- **Edition**: Full
- **JDK Version**: 8u252 RH OpenJDK
- **Operating System**: Linux
- **Database**: None

### PR DESCRIPTION
**Pull request by Cousjava**
## Description
This is a bug fix.

Fixes #4802

### New tests
New test included in PR

### Testing Performed
* Tested using reproducers on Jira tickets
* Microprofile OpenAPI TCK

### Testing Environment
Zulu JDK 1.8_262 on Ubuntu 20.04 with Maven 3.6.3

## Notes for reviewers
Previously schemas may not have been referenced by APIResponse if the schema class was scanned later than the REST class, so this PR means the the APIWalker will go through the classes a second time to add in any references that didn't get linked the first time.

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java [hunk 1]
```diff
@@ -147,5 +147,5 @@ public OpenAPI process(OpenAPI api, OpenApiConfiguration config) {
     private static void removeEmptyPaths(OpenAPI api) {
         PathItem emptyItem = new PathItemImpl();
         api.getPaths().entrySet().removeIf(entry -> emptyItem.equals(entry.getValue()));
-    }
-}
\ No newline at end of file
+    }
```
Comment (dmatej) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java:150]: trailing spaces are ugly ;)

**Comment by Cousjava** [comment]
Fails OpenAPI TCK, needs more work. :(

**Comment by Cousjava** [comment]
Tested and now passes the TCK

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
The stackoverflow error (FISH-390) was fixed already by #4832

**Comment by Pandrex247** [comment]
Tried using reproducer attached to issue FISH-298 - still getting return type of _object_. Can you verify that please? I was under the impression that's not correct and we should be returning an actual impl.

```
...
paths:
  /resources/javaee8/dataviews:
    get:
      summary: Gets descriptions of the dataviews that are available from CDAS.
      operationId: ping
      responses:
        "304":
          description: Not Modified
        "406":
          description: Not Acceptable
        default:
          description: Descriptions of all dataviews.
          content:
            application/json:
              schema:
                type: object
              example: ""
            application/xml:
              schema:
                type: object
...
```

**Comment by Cousjava** [comment]
I get the following with the reproducer:

```
openapi: 3.0.0
info:
  title: Deployed Resources
  version: 1.0.0
servers:
- url: http://JC-VM5-MATE:8080/EE8-1.0-SNAPSHOT
  description: Default Server.
- url: https://JC-VM5-MATE:8181/EE8-1.0-SNAPSHOT
  description: Default Server.
paths:
  /resources/javaee8/dataviews:
    get:
      summary: Gets descriptions of the dataviews that are available from CDAS.
      operationId: ping
      responses:
        "304":
          description: Not Modified
        "406":
          description: Not Acceptable
        default:
          description: Descriptions of all dataviews.
          content:
            application/json:
              schema:
                type: object
                properties:
                  DataviewDescription:
                    type: array
                    description: POJO that represents a DataviewDescription.
                  dataviewDescription:
                    type: array
                    items:
                      $ref: '#/components/schemas/DataviewDescription'
                description: POJO that represents a Dataviews.
              example: ""
            application/xml:
              schema:
                type: object
components:
  schemas:
    DataviewDescription:
      type: object
    Dataviews:
      type: object
      properties:
        DataviewDescription:
          type: array
          description: POJO that represents a DataviewDescription.
        dataviewDescription:
          type: array
          items:
            $ref: '#/components/schemas/DataviewDescription'
      description: POJO that represents a Dataviews.
```
Having no schema for `application/xml` is correct as there is none specifed.

**Review by rdebusscher — APPROVED** [review]
LGTM

**Pull request merged**